DMX.Portal.Web.Tests.Unit/Services/Views/LabViews/LabViewServiceTests.cs

```diff
@@ -0,0 +1,150 @@
+﻿// --------------------------------------------------------------- 
+// Copyright (c) Microsoft Corporation. All rights reserved. 
+// ---------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using DMX.Portal.Web.Brokers.Loggings;
+using DMX.Portal.Web.Models.Labs;
+using DMX.Portal.Web.Models.Labs.Exceptions;
+using DMX.Portal.Web.Models.Views.LabViews;
+using DMX.Portal.Web.Services.Foundations.Labs;
+using DMX.Portal.Web.Services.Views.LabViews;
+using Moq;
+using Tynamix.ObjectFiller;
+using Xeptions;
+using Xunit;
+
+namespace DMX.Portal.Web.Tests.Unit.Services.Views.LabViews
+{
+    public partial class LabViewServiceTests
+    {
+        private readonly Mock<ILabService> labServiceMock;
+        private readonly Mock<ILoggingBroker> loggingBrokerMock;
+        private readonly ILabViewService labViewService;
+
+        public LabViewServiceTests()
+        {
+            this.labServiceMock = new Mock<ILabService>();
+            this.loggingBrokerMock = new Mock<ILoggingBroker>();
+
+            this.labViewService = new LabViewService(
+                labService: this.labServiceMock.Object,
+                loggingBroker: this.loggingBrokerMock.Object);
+        }
+
+        public static TheoryData DependencyExceptions()
+        {
+            var innerException = new Xeption();
+
+            return new TheoryData<Xeption>()
+            {
+                new LabDependencyException(innerException),
+                new LabServiceException(innerException),
+            };
+        }
+
+        private Expression<Func<Xeption, bool>> SameExceptionAs(Xeption expectedException) =>
+            actualException => actualException.SameExceptionAs(expectedException);
+
+        private static List<dynamic> CreateRandomLabViewProperties()
+        {
+            return Enumerable.Range(start: 0, count: GetRandomNumber())
+                .Select(item =>
+                {
+                    LabViewStatus labViewStatus = GetRandomEnum<LabViewStatus>();
+                    LabStatus labStatus = (LabStatus)labViewStatus;

```

Comment:
I believe we should avoid casting the Enums and follow the same (Euclid?) pattern of generating random int and create Enum from that random int.